### DISCUSSION TITLE
Unable to see progress when using batch_run

### DISCUSSION THREAD

**Discussion by ghost** [mesa/mesa#2005] [Q&A]
Below is the part of code where I am using batch_run.
results = mesa.batch_run(
        Abstract_crowdlogistics,
        parameters=DoE,
        iterations=10,
        max_steps=100,
        number_processes=1,
        data_collection_period=1,
        display_progress=True,
        )

results_df = pd.DataFrame(results)

However, when I run this, I do not see the progress of batch_run. Below is a snip of what I see in the console. 

<img width="255" alt="image" src="https://github.com/projectmesa/mesa/assets/131771458/3812d443-77b9-4e19-ae82-50d7321af8d1">

I am using Mesa 2.1.5.

**Comment by rht** [comment]
Can you elaborate on the line `parameters=DoE,` ? Also, it's hard to help debug without access to the code.

**Reply by ghost** [reply]
<img width="536" alt="image" src="https://github.com/projectmesa/mesa/assets/131771458/dc17765a-c8b4-493b-805a-eb0b46e1ab32">

I am trying to create a function which I would call to execute a batch run of my ABM model. I did this so that I can run specific parameters settings (Design of Experiment). 

I also want to add that the model runs and the i am able to read the results_df. The only thing is that I do not see the progress.

**Reply by rht** [reply]
Your `DoE` variable seems to be a static dictionary, not containing a list of possible values. See how it is done in https://github.com/projectmesa/mesa-examples/blob/main/examples/bank_reserves/batch_run.py.

**Reply by ghost** [reply]
Thank you for sharing the example. I see the same issue with the example as seen below:

<img width="764" alt="Screenshot 2024-01-26 at 1 35 47 PM" src="https://github.com/projectmesa/mesa/assets/131771458/fc7c5044-8b2d-4036-8536-a9a9196d40e8">

I think its an issue with the mesa version that I have. I did not see this issue when I was using Mesa 1.2.1. 

I am currently using Mesa 2.1.5.

**Reply by rht** [reply]
It works on my machine
```
100%|██████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:04<00:00,  1.19s/it]
```

Have you tried the latest release, 2.2.4?

**Reply by ghost** [reply]
After I upgraded to 2.2.4, I get the following error:

  File ~/anaconda3/lib/python3.11/site-packages/mesa/datacollection.py:211 in collect
    self._agent_records[model._steps] = list(agent_records)

  File ~/anaconda3/lib/python3.11/site-packages/mesa/datacollection.py:181 in get_reports
    _prefix = (agent.model._steps, agent.unique_id)

AttributeError: 'BankReservesModel' object has no attribute '_steps'

**Reply by rht** [reply]
That means your mesa-examples repo is outdated. Needs to `git pull` to the latest version. The latest version has `super().__init__()` inside the `BankReservesModel` `__init__()` method, but not with the current one you have. If you can confirm this.

**Reply by ghost** [reply]
I checked the code. I does have super().__init__() in the model.

<img width="617" alt="image" src="https://github.com/projectmesa/mesa/assets/131771458/9e93e834-9851-4f30-b1d6-33f9bd7a5b1d">

**Reply by rht** [reply]
OK, have you tried updating to the latest version nonetheless. There has been a new fix since then: https://github.com/projectmesa/mesa-examples/commits/main/examples/bank_reserves.

**Reply by ghost** [reply]
I updated to mesa 2.2.4. I downloaded the latest bank_reserves model and I looked at the "batch_run.py" file. I found that `super().__init__()` was missing in the model. I made the correction in it. I still can not see the progress of the batch run.

<img width="1562" alt="Screenshot 2024-01-28 at 3 14 15 PM" src="https://github.com/projectmesa/mesa/assets/131771458/feca9005-3c48-44d5-940b-b763cccfbf95">

The picture I showed in my previous response was from the "Model.py" file.

I tried to add `super().__init__()` in my research project code and it still does not show me the progress.

**Reply by rht** [reply]
Currently investigating.

**Reply by rht** [reply]
Fixed by https://github.com/projectmesa/mesa-examples/pull/97.
